### ISSUE TITLE
review csrf in cookies ?

### ISSUE THREAD

**Issue by bodik** [snerstack/sner-monorepo#34]
```
    @app.after_request
    def inject_csrf_token(response):
        response.set_cookie('XSRF-TOKEN', generate_csrf())
        return response
```

**Comment by bodik** [comment]
https://github.com/snerstack/sner-monorepo/issues/37

**Comment by bodik** [comment]
csrf token handling in current implementation uses cookies as auto-magic csrf token storage and management for frontend. with each request, token is pushed to storage without any js code. frontend uses cookie value (must not be httponly) to send tokens in headers on POST requests (httpClient axios interceptor, datatables ajax attrs?callbacks).

the header sent has a name not mentioned in docs, but used as alternative value in implementation (https://github.com/wtforms/flask-wtf/blob/main/src/flask_wtf/csrf.py#L199) -- mystery 1 resolved

XSRF-TOKEN cookie is not used by flask_wtf for request validation, only csrf_token form field, or two headers

mattermost uses same implementation, but token cookie name differs in name a lot, perhaps not to collide in any way with arbitrary library implementation ?

perun uses jwt authentication (not session based), uses local storage to store token obtained during login. the token cannot be used by any non-origin code (perhaps depends on csp or other sop mechanisms) and it is not automatically sent with any request as session cookies from CSRF, ?hence extra token is not necessary

**Comment by bodik** [comment]
gitlab is server-side rendered, tokens are present in html responses on various places (meta, javascript handlers) and used from there by browser

**Comment by bodik** [comment]
resolved [b4830fd](https://github.com/snerstack/sner-monorepo/commit/b4830fd10353d070440fb9d47a370a1487e70eb6)
